source/Lib/Utilities/NoMallocThreadPool.h

```diff
@@ -157,7 +157,11 @@ struct BlockingBarrier
     BlockingBarrier* nonconst = const_cast<BlockingBarrier*>(this);
 
     std::unique_lock<std::mutex> l( nonconst->m_lock );
+#if __GNUC__ && (__cplusplus > 201703L)
```

Comment:
Is there a reason to guard the fix with __GNUC__ and CPP version? Why not just make it unconditional?

Reply:
hmm, if vvenc is for c++17 then there is no problem.
Unfortunately I recently added plugin for ffmpeg in gnu++23 and I have warnings.
https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0806r2.html

Reply:
The question is. Does the code compile in c++17? If yes, there is no need to use two version of this code.

Reply:
As mentioned, please remove the guard and just change the code, unless its not c++17 compliant.

Reply:
There is absolutely no reason to check `__MINGW32__ or `__GNUC__` here. Nothing about the code is GCC-specific.